app-agent/src/exec_process.rs

```diff
@@ -33,6 +54,198 @@ pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result
     Ok(status)
 }
 
+fn handle_permission_denied(external_command: String) -> String {
+    let file_permission_denied = match File::open(external_command.clone()) {
+        Ok(file) => file,
+        Err(err) => {
```

Comment:
the `Ok` branch is very small, you can turn this into a `if let` instead. Also, what does `file_permission_denied` represent?

It would be cleaner to make it a separate function.